utilities/data-ingestion/ingest-institutions.py

```diff
@@ -0,0 +1,221 @@
+import boto3
+import json
+from decimal import Decimal
+from datetime import datetime
+
+
+s3_client = boto3.client("s3")
+sqs = boto3.client("sqs")
+
+
+def lambda_handler(event, context):
+    # Helper function to convert to Decimal, has to have an
+    # option to return original string
+    def convertToNumeric(inputString, isInteger=False):
+        if len(inputString) == 0:
+            return None
+        try:
+            if isInteger:
+                return Decimal(int(inputString))
+            return Decimal(inputString)
+        finally:
+            return inputString
+
+    try:
+        # Grab file from S3
+        s3_Bucket_Name = event["Records"][0]["s3"]["bucket"]["name"]
+        s3_File_Name = event["Records"][0]["s3"]["object"]["key"]
+        object = s3_client.get_object(Bucket=s3_Bucket_Name, Key=s3_File_Name)
+
+        # massage data
+        data = object["Body"].read().decode("utf-8")
+        institutions = data.split("\n")
+        institutions.pop(0)  # removes header row
+        records_processed = 0
+
+        # transform data into Item and add to queue
+        for institution in institutions:
+            institution_data = institution.split(",")
+            try:
+                sqs.send_message(
+                    QueueUrl="https://sqs.us-east-1.amazonaws.com/905418154281/process-institutions",
+                    MessageBody=json.dumps(
+                        {
+                            "institutionId": int(institution_data[0]),
+                            "recordType": "data",
+                            "updatedAt": datetime.now().strftime(
+                                "%Y-%m-%dT%H:%M:%S.%f"
+                            ),
+                            "institutionName": institution_data[1],
+                            "city": institution_data[3],
+                            "state": institution_data[4],
+                            "zip": institution_data[5],
+                            "latitude": convertToNumeric(institution_data[11]),
+                            "longitude": convertToNumeric(institution_data[12]),
+                            "url": institution_data[7],
+                            "control": convertToNumeric(institution_data[6], True),
```

Comment:
could consider renaming this to `institutionType` or something more self-explanatory. Also would be nice if we used the text instead of the numbers here so we didn't have to map it

Reply:
Will do, standby please

Reply:
<img width="449" alt="image" src="https://github.com/coforma/swift-tech-challenge/assets/7631428/9ada58ab-4551-44ff-a9d1-38a9021d18f1">

Reply:
The data is in the DB, will revise the PR again